scripts/zones/Uleguerand_Range/npcs/Buffalostalker_Dodzbraz.lua

```diff
@@ -1,22 +1,34 @@
 -----------------------------------
 -- Area: Uleguerand Range
---  NPC: Buffalostalker Dodzbraz
+-- NPC: Buffalostalker Dodzbraz
 -- Type: Quest NPC
+-- Quest: Bombs Away! (96)
 -- !pos -380.171 -24.89 -180.797 5
+-- !additem 1667 (Cluster Core)
+-- !additem 5267 (Shumeyo Salt)
 -----------------------------------
 local entity = {}
 
 entity.onTrade = function(player, npc, trade)
+    if npcUtil.tradeHas(trade, { { xi.items.CLUSTER_CORE, 2 } }) then -- Cluster Core x2
+        player:startEvent(8, 1667)
+    else
+        player:startEvent(7)
+    end
 end
 
 entity.onTrigger = function(player, npc)
-    player:startEvent(6)
+    player:startEvent(6, 1667)
 end
 
 entity.onEventUpdate = function(player, csid, option)
 end
 
 entity.onEventFinish = function(player, csid, option)
+    if csid == 8 then
```

Comment:
(on mobile) 

The logic here won't actually work in some cases. GiveItem returns true/false (I think) depending on if the item was successfully given or not. It can fail if the player has a full inventory. From memory: we don't account for the contents of the trade being taken to free up space for reward items, so it should look like:

```
if csid == 8 and npcutils.giveitem(...) then
    confirm trade 
end
```

Would be useful to see what retail does when the trade fails after being given relevant items but not having enough space.